# Sok-2030 Næringsøkonomi og konkuransestrategi

## Monopol og prisdiskriminering


In [1]:
import sympy as sp
import numpy as np
import matplotlib.pyplot as plt

### Grunnleggende Monopolmodell

Monopolisten står overfor en etterspørselskurve og maksimerer profitt ved å sette marginalinntekten lik marginalkostnaden (MR = MC).

Men før vi kan sette marginalinntekten lik marginalkostnaden, må vi finne marginalinntekten og marginalkostnaden. Disse blir normalt gitt ut i oppgavene, men kort forklart er det den deriverte av inntektsfunksjonen og kostnadsfunksjonen.

Litt repitisjon før vi begynner på monopol og prisdiskriminering:

## Forklaring av variabler i monopolmodellen

For å forstå monopol og prisdiskriminering må vi først definere de viktigste variablene:

- $P$ er prisen monopolisten setter for produktet sitt.
- $Q$ er mengden av produktet som produseres totalt.
- $MR$ er marginalinntekt: Inntekten monopolisten får ved å selge én ekstra enhet.
- $MC$ er marginalkostnad: Kostnaden for å produsere én ekstra enhet.
- $\pi $ er (Profitt): Totale inntekter minus totale kostnader.
- $q$ er kvantum: Om det er flere fabrikker, vil $q$ være mengden produsert av én fabrikk.

### Etterspørselsfunksjon:
En lineær invers etterspørselsfunksjon har formen:

$$P(Q) = a - bq$$

Der:
- $a$ representerer den maksimale betalingsvilligheten (når $Q = 0$).
- $b$ angir hvor følsom prisen er for endringer i kvantum.

In [2]:
# Definerer symboler
a,b,Q,P, F,c,d,q = sp.symbols('a b Q P F c d q', real=True, positive=True)

# Definerer funksjonen for etterspørsel
P_eq = a - b*q

### Marginalinntekt:

I en monopolistisk setting er den **totale inntekten (TR)** definert som produktet av prisen $P(q)$ og kvantumet $q$:

$$
TR(Q) = P(q) \cdot q
$$

Ved å sette inn $P(Q)$ i uttrykket for $TR(Q)$ får vi:

$$
TR(Q) = (a - bq) \cdot q 
$$

og deretter forenklet blir:

$$
TR(Q) = aq - bq^2
$$

In [3]:
# For å finne totalinntekten, multipliserer vi prisen med kvantumet
TR = P_eq*q
TR

q*(a - b*q)

### Derivering av totale inntekter

Marginalinntekten (MR) er definert som den deriverte av de totale inntektene med hensyn til $Q$:

$$
MR(Q) = \frac{\partial TR(Q)}{\partial Q}
$$

Vi deriverer $TR(Q) = aQ - bQ^2$ med hensyn til $Q$:

1. Den deriverte av $aQ$ med hensyn til $Q$ blir $a$.
2. Den deriverte av $bQ^2$ med hensyn til $Q$ blir $2bQ$.

Så får vi:

$$
MR(Q) = a - 2bQ
$$

Det vi kan se er at:

- Når $Q$ øker, reduseres marginalinntekten med en hastighet på $2b$ per enhet økning i $Q$.
- Marginalinntekten faller raskere enn prisen, noe som er en viktig egenskap i en monopolistisk markedsstruktur.

In [4]:
# Deriverer totalinntekten med hensyn på kvantumet og finner marginalinntekten
MR = sp.diff(TR, Q)
MR

0

##  Marginalkostnad

**Marginalkostnad (MC)** er definert som endringen i totale kostnader når produksjonsmengden økes med én enhet. Det vil si at marginalkostnaden viser hvor mye den totale kostnaden øker for hver ekstra produsert enhet.

La oss anta at de **totale kostnadene (TC)** er gitt ved en funksjon:
$$
TC(q) = F + cq + dq^2
$$
hvor $q$ er antall produserte enheter i én fabrikk.

Dette uttrykket forteller oss hvor mye de totale kostnadene endres når vi øker $q$ med en enhet.
hvor:
- $F$ er de faste kostnadene (kostnader som ikke endres med produksjonsmengden),
- $c$ er den variable kostnaden per enhet, og
- $dq^2$ representerer økende variable kostnader (f.eks. på grunn av kapasitetsskala eller andre ineffektivitetseffekter ved høy produksjon).

Merk at om monopolisten har flere fabrikker, vil $q$ være kvantum produsert av én fabrikk, og $Q$ vil være totalt kvantum produsert av alle fabrikkene.

Da blir uttrykket for total kostnad:

$$
TC(Q) = F + cQ + dQ^2
$$

hvor $q$ er $ \frac{Q}{n}$ kvantum produsert mellom alle fabrikkene.

In [5]:
# Definerer funksjonen for totalkostnaden, q = Q/2 hvor monopolisten fordeler kvantum produsert på 2 fabrikker
q = Q/2 
TC = F + c*q + d*q**2
TC

F + Q**2*d/4 + Q*c/2

Marginalkostnaden defineres som den deriverte av de totale kostnadene med hensyn til produksjonsmengden:
$$
MC(Q) = \frac{\partial TC(Q)}{\partial Q}
$$

For å finne marginalkostnaden, deriverer vi $TC(Q)$ med hensyn til $Q$:
1. Den deriverte av $F$ (en konstant) er 0.
2. Den deriverte av $\frac{Q^{2} d}{4}$ med hensyn til $Q$ er $\frac{2Qd}{4}$.
3. Den deriverte av $\frac{Q c}{2}$ med hensyn til $Q$ er $\frac{c}{2}$.

Så får vi:
$$
MC(Q) =  \frac{Q d}{2} + \frac{c}{2}
$$

Dette uttrykket viser at marginalkostnaden øker med $q$ dersom $d > 0$, noe som kan være typisk i mange produksjonsscenarier.

In [6]:
MC = sp.diff(TC, Q)
MC

Q*d/2 + c/2

# Eksempel

La oss si at vi har en monopolist som selger en vare til en pris gitt ved lineær etterspørsel fra 2.1.2 monopoltilpasning:

$$
Q = 6000 - \frac{50P}{9}
$$

Vi løser for $P$ og får invers etterspørsel:

$$
P(Q) = \frac{6000}{50} - \frac{9Q}{50}
$$

Forenkler litt med å dele 6000 på 50 og får:

$$
P(Q) = 120 - \frac{9Q}{50}
$$

In [7]:
# Gitt etterspørselsfunksjon: Q = (6000 - 50*P)/9
ettersporsel = sp.Eq(Q, (6000 - 50*P)/9)

# Løs for P for å finne den inverse etterspørselen
invers_etters = sp.solve(ettersporsel, P)[0]

invers_etters

120 - 9*Q/50

Deretter finner vi marginalinntekt ved å multiplisere $P(Q)$ med $Q$ og deriverer med hensyn til $Q$:

$$
TR(Q) = (120 - \frac{9Q}{50}) \cdot Q
$$

Multipliserer ut og får:

$$
TR(Q) = 120Q - \frac{9Q^2}{50}
$$

Og deriverer med hensyn til $Q$:

$$
MR(Q) = 120 - 2 \cdot \frac{9Q}{50}
$$

Så har vi marginalkostnaden:

$$
MR(Q) = 120 - \frac{18Q}{50}
$$

In [8]:
# Multipliserer prisen med kvantumet for å finne totalinntekten
TR = invers_etters*Q
TR

Q*(120 - 9*Q/50)

In [9]:
# Deriverer totalinntekten med hensyn på kvantumet for å finne marginalinntekten
MR = sp.diff(TR, Q)

# Python forenkler alltid uttrykkene, så om de ser litt rare ut, er det fordi de er forenklet
# 120 - 9Q/25 er det samme som 120 - 18Q/50
MR

120 - 9*Q/25

## Finner marginalkostnaden

De totale kostnadene er gitt ved:

$$
TC(q) = 100 + q^2 + 10q
$$

Merk at i oppgave 2.1.2 når hele konsernet produserer totalmengden Q fordeler den seg likt på 50 fabrikker, altså
$$
q = \frac{Q}{50}
$$

per fabrikk.

Total kostnad er 50 ganger kostnaden i én fabrikk, men hver fabrikk produserer $ \frac{Q}{50} $ enheter. Da får vi:

$$
TC(Q) = 50 \cdot [100 + (\frac{Q}{50})^2 + (\frac{10Q}{50})]
$$

Med å regne ut ledd inne i klammen først får vi:

$$
TC(Q) = 50 \cdot 100 + \frac{Q^2}{2500} + \frac{Q}{5}
$$

Multipliserer ut med 50 og får:

$$
TC(Q) = 5000 + \frac{Q^2}{50} + 10Q
$$



For å finne marginalkostnaden, deriverer vi $TC(Q)$ med hensyn til $Q$:
1. Den deriverte av $5000$ (en konstant) er 0.
2. Den deriverte av $\frac{Q^2}{50}$ med hensyn til $Q$ er $\frac{2Q}{50}$.
3. Den deriverte av $10Q$ med hensyn til $Q$ er $10$.

Så får vi:
$$
MC(Q) = \frac{2Q}{50} + 10
$$

In [10]:
# Definerer total kostnad og at q = Q/50
q = Q/50
TC = 100 + q**2 + 10*q
TC

Q**2/2500 + Q/5 + 100

In [11]:
# Vi har 50 fabrikker og det er monopolistisk konkurranse
# Derfor er det 50 bedrifter som produserer kvantumet q
TC = 50*TC
TC

Q**2/50 + 10*Q + 5000

In [12]:
# Deriverer total kostnad med hensyn på kvantumet
# Utrykket er forenklet av python
MC = sp.diff(TC, Q)
MC

Q/25 + 10

# Optimal tilpasning for monopolisten

For å finne optimal tilpasning for monopolisten, må vi sette marginalinntekten lik marginalkostnaden, disse har vi nå regnet ut:

$$
MR(Q) = MC(Q)
$$

Setter vi inn uttrykkene for $MR(Q)$ og $MC(Q)$ får vi:

$$
120 - \frac{18Q}{50} = 10 + \frac{Q}{25}
$$

Løser vi denne likningen for Q får vi:

$$
Q = 500
$$

Setter vi inn $Q = 500$ i etterspørselsfunksjonen får vi:

$$
P = 120 - \frac{9 \cdot 500}{50} = 70.5
$$

Så monopolisten bør produsere 500 enheter og selge dem til en pris på 70.5.

In [13]:
# Setter MR = MC for å finne likevektskvantumet
likevekt = sp.Eq(MR, MC)

# Løser for Q
Q_løsning = sp.solve(likevekt, Q)

# Setter inn kvantumet i etterspørselsfunksjonen for å finne prisen
P_løsning = invers_etters.subs(Q, Q_løsning[0])

f'Q blir {Q_løsning[0]}, P blir {round(P_løsning, 1)}'


'Q blir 275, P blir 70.5'

## Produksjon per fabrikk og profitt

Siden vi vet at det er 50 fabrikker, og hver fabrikk produserer $ \frac{Q}{50} $ enheter hvor Q er 275, får vi:

$$
q = \frac{275}{50} = 5.5 
$$

per fabrikk.

For å finne profitten til monopolisten, trekker vi fra de totale kostnadene fra de totale inntektene:

Total inntekt er gitt ved:

$$
\pi(q) = P \cdot q - TC(q)
$$

Setter vi inn $P = 70.5$, $q = 5.5$, og $TC(q) = 100 + 5.5^2 + 10 \cdot 5.5$ i profittfunksjonen får vi:

$$
\pi(q) = 70.5 \cdot 5.5 - 100 + 5.5^2 + 10 \cdot 5.5 = 202.5
$$

Produksjon per fabrikk er 5.5 enheter og profitten per fabrikk er 202.5.

In [14]:
# Regner ut produksjon per fabrikk og profitt
Optimal_q= round(Q_løsning[0]/50, 2)
Profitt_per_fabrikk = (round(P_løsning,2) * Optimal_q) - (100 + Optimal_q**2 + 10*Optimal_q)


f'Produksjon per fabrikk blir {Optimal_q} og profitt per fabrikk blir {Profitt_per_fabrikk}'

'Produksjon per fabrikk blir 5.50 og profitt per fabrikk blir 202.5'

## Optimal tilpasning ved tredje grads prisdiskriminering